unreal: clarify environment defaults (#11991)
getsentry/sentry-docs: docs/platforms/unreal/configuration/environments.mdx

---

> Environments tell you where an error occurred, whether that's in your production system, your staging server, or elsewhere.

Reviewer: ```suggestion
Environments in Sentry let you know where an error occurred, (such as in production, staging server, or another location).
```

> It help you better filter issues, releases, and user feedback in the Issue Details page of sentry.io, which you learn more about in our [documentation that covers using environments](/product/sentry-basics/environments/).

Reviewer: ```suggestion
They make it easier to filter issues, releases, and user feedback on the Issue Details page, (which you can navigate to by clicking any issue on the [**Issues**](https://sentry.io/orgredirect/organizations/:orgslug/issues/) page). To learn more, read our docs about [using environments](/product/sentry-basics/environments/).
```

Sentry automatically creates an environment when it receives an event with the `environment` parameter set.

By default, the SDK reports `Editor` when running inside the Unreal Engine Editor. 

> For shipping builds it'll use `Release` and in other configurations it'll use `FApp::GetBuildConfiguration()`

Reviewer 2: ```suggestion
For shipping builds it'll use `Release` and in other configurations it'll use `FApp::GetBuildConfiguration()`.
```

> By default, the SDK reports `Editor` when running inside the Unreal Engine Editor. 
> For shipping builds it'll use `Release` and in other configurations it'll use `FApp::GetBuildConfiguration()`

Reviewer: ```suggestion
When running inside the Unreal Engine Editor, the SDK reports the environment as `Editor`. For shipping builds, it defaults to `Release`. In other configurations it uses the `FApp::GetBuildConfiguration()`. 
```

> You can also set the default environment to `Release`, `Development`, or `Debug` in your build configuration or override the environment value programatically.

Reviewer: ```suggestion
You can also set the default environment to `Release`, `Development`, or `Debug` in your build configuration, or override it programmatically.
```

SentrySubsystem->InitializeWithSettings(SettingsDelegate);
```

> Environments are case-sensitive. The environment name can't contain newlines, spaces or forward slashes, can't be the string "None", or exceed 64 characters. You can't delete environments, but you can [hide](/product/sentry-basics/environments/#hidden-environments) them.

Reviewer: ```suggestion
**Important things to note about environments:**

- Environment names are case-sensitive.
- Names cannot include new lines, spaces, or forward slashes.
- Certain strings, like "None", aren't allowed.
- The maximum length for an environment name is 64 characters.
- Environments can't be deleted, but can be [hidden](/product/sentry-basics/environments/#hidden-environments) if no longer needed.

```
